themes/Backend/ExtJs/backend/config/view/form/document.js

```diff
@@ -98,6 +98,23 @@ Ext.define('Shopware.apps.Config.view.form.Document', {
                 fieldLabel: '{s name=document/detail/name_label}Name{/s}',
                 allowBlank: false,
                 translatable: true
+            },{
+                name: 'key',
+                fieldLabel: '{s name=document/detail/key_label}Internal Identifier{/s}',
+                allowBlank: false,
+                validator: function(key){
+                    var existingKeys = [];
+
+                    Ext.getStore('form.Document').each(function(record) {
+                        existingKeys.push(record.data.key);
+                    });
+
+                    if (existingKeys.indexOf(key) === -1 && key !== '') {
+                        return true;
+                    } else {
```

Comment:
I think you should give a different error message depending on which of the checks above has failed. This one is nice when no key was specified, but in cases where the key collides with another key, you should give one that spells out the problem more explicitly.